server/src/server.ts

```diff
@@ -543,6 +543,24 @@ function fixCompletion(c: CompletionItem, newText: string, replaceStart: Positio
 	}
 }
 
+async function getTokenData(uri : string, position : Position): Promise<[FixedValidationResult, TextDocument, CallChain[]] | null> {
```

Comment:
this looks much harder then it needs to be

Reply:
It is just a common code which is used in almost every `connection` callback, so I thought it would be useful to have it in one place

Reply:
no, in onCompletion another code